assembly/models/extrinsic/generic-extrinsic.ts

```diff
@@ -0,0 +1,120 @@
+import { Bool, Byte, BytesReader, Codec, CompactInt, UInt64 } from "as-scale-codec";
+import { Utils } from "../../utils";
+import { TransactionTag, TransactionValidity, ValidTransaction } from "../transaction-models";
+import { Call } from "./call";
+import { ExtrinsicSignature } from "./extrinsic-signature";
+import { Transfer } from "./transfer";
+
+export enum TransactionSource {
+    InBlock = 0,
+    Local = 1,
+    External = 2,
+}
+
+export class GenericExtrinsic<Address extends Codec, B extends Codec, N extends Codec, S extends Codec> implements Codec {
+    private _method: Call;
+    private _signature: ExtrinsicSignature<Address, S>;
+    static readonly SIGNING_BIT_SET: u8 = 132;
+
+    constructor(
+        method: Call = new Call(), 
+        signature: ExtrinsicSignature<Address, S> = new ExtrinsicSignature())
+    {
+        this._method = method;
+        this._signature = signature;
+    }
+
+    public get method(): Call {
+        return this._method;
+    }
+
+    public get signature(): ExtrinsicSignature<Address, S> {
+        return this._signature;
+    }
+
+    /**
+     * @description The payload being signed in transactions.
+     */
+    getPayload(): u8[] {
+        return this.method.toU8a()
+            .concat(this.signature.signedExtension.toU8a());
+    }
+
+    isSigned(): bool {
+        return !this.signature.isEmpty;
+    }
+
+    asTransfer(): Transfer<Address, B, N> {
```

Comment:
I think that this is logic specific to a pallet (in this case `Ballance`). IMO this logic should be generated by the dispacher in order to call the balances pallet with the transfer information. The generic extrinsic responsibility is only to provide the call (module + index) + the arguments